### ISSUE TITLE
`navigateTo` with `open` and `name` params opens a home url.

### ISSUE THREAD

**Issue by peterkimzz** [nuxt/nuxt#28130]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.13.1
- Nuxt Version:     3.11.2
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  yarn@1.22.22
- Builder:          -
- User Config:      modules, devtools, nitro, image, colorMode, ui, tailwindcss, eslint, typescript, supabase, runtimeConfig, routeRules, vite, security
- Runtime Modules:  nuxt-security@1.0.1, @nuxt/devtools@1.3.6, @nuxtjs/device@3.1.1, @nuxt/content@2.13.0, @nuxtjs/supabase@1.1.7, @nuxt/ui@2.16.0, @nuxt/eslint@0.3.13, @nuxt/image@1.7.0, @nuxtjs/i18n@8.3.1
- Build Modules:    -
------------------------------

### Reproduction

```ts
const ROUTE_NAMES = {
  LOGIN: "login"
}

navigateTo({ name: ROUTE_NAMES.LOGIN }, { open: { target: "_blank" } });
```

### Describe the bug

Hi,

When I use `navigateTo` with `name` and `open: "_target"`, open the home url.

**Comment by divagueame** [comment]
I tried to reproduce the described by bug it seems to work for me.

https://github.com/divagueame/nuxtbugreport

![nuxt-succes](https://github.com/user-attachments/assets/65659716-d5be-4199-a416-0634bdfca0fb)

**Comment by piotr-skowronski-prpl** [comment]
@peterkimzz It's been fixed in `3.12.3`

https://github.com/nuxt/nuxt/pull/27742

**Comment by manniL** [comment]
Fixed via https://github.com/nuxt/nuxt/pull/27742
